README.md

```diff
@@ -83,3 +83,17 @@ You can find contributing guide [here](./CONTRIBUTING.md).
 
 ### Building & deploying locally
 Learn how to build & deploy KEDA locally [here](./BUILD.md).
+
+## Compatibility matrix
```

Comment:
As discussed, please move this to our Kubernetes compatibility doc instead of here to consolidate things and avoid duplication